design/open/acl/rbac-to-acl.md

```diff
@@ -0,0 +1,204 @@
+is more than one concurrent lakeFS).  We force admins to try the script on a
+dry run using a `--yes` flag: without it, the script only prints the planned
+changes.
+
+The upgrade script will _ensure_ that the 4 default global groups exist, and
+_modify_ existing groups to fit into the new ACLs scheme.
+
+* When creating the 4 default global groups: if another group exists and has
+  the desired name, upgrading will rename it by appending ".orig".  So after
+  upgrading the 4 default global groups exist, with these known names.
+* For any group, upgrading configured policies follows these rules, possibly
+  **increasing** access:
+
+  1. Any "Deny" rules are stripped, and a warning printed.
+  1. "Manage own credentials" is added.
+  1. If any actions outside of "fs:" are allowed, the group becomes an Admin
+     group, a warning is printed, and no further changes apply.
+  1. The upgrade script *unifies* repositories: If a permission applies to a
+     set of repositories with a wildcard, permissions are unified to **all**
+     repositories.  Otherwise they apply to the list of all repositories, in
+     all the policies.
+  1. The upgrade script *unifies* actions: it selects the _least_ permission
+     of Read, Write, Super that contains all of the allowed actions.  If any
+     allowed action contains a wildcard this still makes sense.
+
+  All warnings link to support@ and to Slack.
+* The upgrade script prints the new permissions of the group.[^1]
+
+
+[^1] Even JSON-diff cannot be used report detailed changes, as many policies
+	 are unified into a single policy.
+
+### PBAC user experience with external authorization server
+
+If lakeFS is attached to an external authorization server, the existing PBAC
+GUI may continue to be used.  It is not possible to use both types of GUI at
+the same time: Moving from PBAC to simplified may only be performed once and
+**will** lose configuration.
+
+## Implementation
+
+lakeFS enforcement will remain backed by PBAC.  We shall add a "ui" field to
+policies.  It will contain a structure that backs the UI:
+
+<table>
+<tr><th>Read all repos</th><th>Read repos `foo`, `bar`</th></tr>
+<tr><td>
+```json
+{
+  "statement": [
+    {
+      "action": ["fs:List*", "fs:Read*"],
+      "effect": "allow",
+      "resource": "*"
+    }
+  ],
+  "ui": {
+    "permission": "Read",
+    "repositories": { "all": true }
+  }
+}
+```
+</td><td>
+```json
+{
+  "statement": [
+    {
+      "action": ["fs:List*", "fs:Read*"],
+      "effect": "allow",
+      "resource": "foo"
+    }, {
+      "action": ["fs:List*", "fs:Read*"],
+      "effect": "allow",
+      "resource": "bar"
+    }
+  ],
+  "ui": {
+    "permission": "Read",
+    "repositories": { "list": ["foo", "bar"] }
+  }
+}
+```
+</td></tr>
+</table>
+
+The UI receives and edits only the ui sub-object.  lakeFS translates it back
+to a statement when the UI posts it back.  In effect PBAC statements are the
+materialized form of the ui sub-object.
+
+## Milestones
```

Comment:
How the transition will be - I mean we commit everything after all is tested outside main?
Do we plan to deploy some functionality and disable some as we go between milestones?

Reply:
No, the whole thing drops as a single release.  That release will delete P/RBAC and replace it with ACLs.